core/src/main/java/software/amazon/awssdk/auth/ProfileCredentialsProvider.java

```diff
@@ -70,7 +71,13 @@ public static Builder builder() {
 
     @Override
     protected AwsCredentials loadCredentials() {
-        return profilesConfigFile == null ? null : profilesConfigFile.getCredentials(profileName);
+        if (profilesConfigFile == null) {
+            throw new SdkClientException("Unable to find profile configuration file. If you are specifying a profile config "
+                                         + "please verify it exists. If you are specifying a file location provider, please "
```

Comment:
Overriding the location provider is just a test API, so we can probably exclude it here.

Reply:
Updated